src/listener/can_open_listener.py

```diff
@@ -96,10 +100,25 @@ def set_interpreter(self, interpreter):
 
         self.interpreter = interpreter
 
+    def _check_complex_variable(self, node_sdo, index):
+        for subindex in node_sdo[index]:
+            # Skips subindex 0 because there are no value changes around this
+            if(subindex != 0):
+                index_and_subindex = f'{index}sub{subindex}'
+                sdo_value = node_sdo.upload(index, subindex)
+                # Checks for every subindex if value changed
+                if(self._sdo_value_changed(index_and_subindex, sdo_value)):
+                    self.inform_interpreter(index_and_subindex)
+
+    def _check_simple_variable(self, node_sdo, index):
+        sdo_value = node_sdo.upload(index, 0)
+        if(self._sdo_value_changed(index, sdo_value)):
+            self.inform_interpreter(index)
+
```

Comment:
Moeten deze niet iets returnen?

Reply:
Wat zou het moeten returnen dan?

Reply:
Benaming aangepast om enige verwarring te vermijden

Reply:
zou je dit niet "read" ipv "check" noemen?

Reply:
dat is toch al gebeurd? https://github.com/ZeroPoint095/CargoDash/pull/23/commits/39ead6162f5686b642d04536daf75d98a96184f8